### PR TITLE
fix(eip/associate): add state refresh block and refactor the test content

### LINKED ISSUE #2254: [EIP] ECS instance cannot obtain the IP address in time
Hi team,

I got an issue of the EIP associate:
After the EIP associate is completed, the ECS instance cannot obtain the IP address in time (>70%).

Test steps:
1. create the ECS instance and bind an EIP.
2. query the IP address from ECS data source.
3. get an empty value.
4. retry.
5. get the IP address.

Here is my script:
```
data "huaweicloud_availability_zones" "default" {}

data "huaweicloud_compute_flavors" "default" {
  availability_zone = data.huaweicloud_availability_zones.default.names[0]
  performance_type  = "normal"
  cpu_core_count    = 8
  memory_size       = 16
}

data "huaweicloud_images_image" "default" {
  name        = "Ubuntu 18.04 server 64bit"
  most_recent = true
}

resource "huaweicloud_vpc" "default" {
  name = var.vpc_name
  cidr = "192.168.0.0/16"
}

resource "huaweicloud_vpc_subnet" "default" {
  name        = var.subnet_name
  cidr        = "192.168.0.0/24"
  gateway_ip  = "192.168.0.1"
  vpc_id      = huaweicloud_vpc.default.id
  ipv6_enable = true
}

resource "huaweicloud_networking_secgroup" "default" {
  name                 = var.security_group_name
  delete_default_rules = true
}

resource "huaweicloud_compute_instance" "default" {
  name               = var.ecs_instance_name
  image_id           = data.huaweicloud_images_image.default.id
  flavor_id          = data.huaweicloud_compute_flavors.default.ids[0]
  availability_zone  = data.huaweicloud_availability_zones.default.names[0]
  # key_pair           = huaweicloud_compute_keypair.default.name
  security_group_ids = [huaweicloud_networking_secgroup.default.id]

  admin_pass = "Overlord!!52259"

  network {
    uuid = huaweicloud_vpc_subnet.default.id
  }
}

resource "huaweicloud_vpc_eip" "default" {
  publicip {
    type = "5_bgp"
  }

  bandwidth {
    share_type  = "PER"
    size        = 5
    name        = var.bandwidth_name
    charge_mode = "traffic"
  }
}

resource "huaweicloud_vpc_eip_associate" "default" {
  public_ip  = huaweicloud_vpc_eip.default.address
  network_id = huaweicloud_compute_instance.default.network[0].uuid
  fixed_ip   = huaweicloud_compute_instance.default.network[0].fixed_ip_v4
}

data "huaweicloud_compute_instance" "default" {
  depends_on = [huaweicloud_vpc_eip_associate.default]

  name = var.ecs_instance_name
}

output "public_ip_address_output" {
  value = data.huaweicloud_compute_instance.default.public_ip
}
```
The result is:
```
huaweicloud_compute_instance.default: Creation complete after 1m8s [id=4d599f81-36d8-474b-ae35-09a953e3c7ce]
huaweicloud_vpc_eip_associate.default: Creating...
huaweicloud_vpc_eip_associate.default: Creation complete after 8s [id=a4d53015-9fc5-49a5-86f3-7e0cd23e88aa]
data.huaweicloud_compute_instance.default: Reading...
data.huaweicloud_compute_instance.default: Read complete after 1s [id=4d599f81-36d8-474b-ae35-09a953e3c7ce]

Apply complete! Resources: 6 added, 0 changed, 0 destroyed.

Outputs:

public_ip_address_output = ""
```
`huaweicloud_compute_eip_associate` has the same problem:
```
resource "huaweicloud_compute_eip_associate" "default" {
  public_ip   = huaweicloud_vpc_eip.default.address
  instance_id = huaweicloud_compute_instance.default.id
}

data "huaweicloud_compute_instance" "default" {
  # depends_on = [huaweicloud_vpc_eip_associate.default]
  depends_on = [huaweicloud_compute_eip_associate.default]

  name = var.ecs_instance_name
}
```

Please help to analysis.

### PR DESCRIPTION
**Pull request by Lance52259**
**What this PR does / why we need it**:
This PR solves the problem of IP query failure after EIP binding:
- after EIP associate is completed, the ECS instance cannot obtain the IP address in time.

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #2254 

**Special notes for your reviewer**:

**Release note**:

```release-note
1. rename resource filename.
2. add the status waiting block.
3. refactor the test content.
```

## PR Checklist

* [x] Tests added/passed.
* [ ] Documentation updated.
* [ ] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud' TESTARGS='-run=TestAccComputeEIPAssociate_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run=TestAccComputeEIPAssociate_basic -timeout 360m -parallel 4
=== RUN   TestAccComputeEIPAssociate_basic
=== PAUSE TestAccComputeEIPAssociate_basic
=== CONT  TestAccComputeEIPAssociate_basic
--- PASS: TestAccComputeEIPAssociate_basic (194.22s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       194.309s
```
```
make testacc TEST='./huaweicloud/services/acceptance/eip' TESTARGS='-run=TestAccEIPAssociate'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/eip -v -run=TestAccEIPAssociate -timeout 360m -parallel 4
=== RUN   TestAccEIPAssociate_basic
=== PAUSE TestAccEIPAssociate_basic
=== RUN   TestAccEIPAssociate_port
=== PAUSE TestAccEIPAssociate_port
=== RUN   TestAccEIPAssociate_compatible
=== PAUSE TestAccEIPAssociate_compatible
=== CONT  TestAccEIPAssociate_basic
=== CONT  TestAccEIPAssociate_compatible
=== CONT  TestAccEIPAssociate_port
--- PASS: TestAccEIPAssociate_port (87.24s)
--- PASS: TestAccEIPAssociate_compatible (88.56s)
--- PASS: TestAccEIPAssociate_basic (224.96s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/eip       225.025s
```

### CODE REVIEW

File: huaweicloud/resource_huaweicloud_compute_eip_associate.go [hunk 1]
```diff
@@ -93,11 +118,10 @@ func resourceComputeEIPAssociateCreate(d *schema.ResourceData, meta interface{})
 	}
 
 	// get port id
-	portID, privateIP, err := getComputeInstancePortIDbyFixedIP(d, config, instanceID, fixedIP)
+	vpcId, portID, privateIP, err := getComputeInstancePortIDbyFixedIP(ecsClient, config, instanceID, fixedIP)
```
Comment (ShiChangkuo) [inline huaweicloud/resource_huaweicloud_compute_eip_associate.go]: I think `vpcId` is not necessary to get the public IP, you can refer to computePublicIP function.

File: huaweicloud/services/state_refresh.go [hunk 2]
```diff
@@ -0,0 +1,17 @@
+package services
+
+import (
+	"time"
+)
+
+// CalculatePollInterval is a method to calculate the query interval time based on the maximum waiting time.
+// Timeout:   5m 10m 15m 20m 25m 30m 35m 40m 45m 50m 55m 60m
+// Interval: 10s 10s 12s 13s 14s 16s 17s 18s 20s 21s 22s 24s
+func CalculatePollInterval(t time.Duration) time.Duration {
```
Comment (ShiChangkuo) [inline huaweicloud/services/state_refresh.go]: I think you should explain the purpose of designing this feature before the PR

File: huaweicloud/resource_huaweicloud_compute_eip_associate.go (hunk 1 above)
Comment (Lance52259) [reply huaweicloud/resource_huaweicloud_compute_eip_associate.go]: done.

File: huaweicloud/services/state_refresh.go (hunk 2 above)
Comment (Lance52259) [reply huaweicloud/services/state_refresh.go]: done.

**Comment by Lance52259** [comment]
```
make testacc TEST='./huaweicloud' TESTARGS='-run=TestAccComputeEIPAssociate_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud -v -run=TestAccComputeEIPAssociate_basic -timeout 360m -parallel 4
=== RUN   TestAccComputeEIPAssociate_basic
=== PAUSE TestAccComputeEIPAssociate_basic
=== CONT  TestAccComputeEIPAssociate_basic
--- PASS: TestAccComputeEIPAssociate_basic (186.62s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       186.931s
```
```
make testacc TEST='./huaweicloud/services/acceptance/eip' TESTARGS='-run=TestAccEIPAssociate_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/eip -v -run=TestAccEIPAssociate_basic -timeout 360m -parallel 4
=== RUN   TestAccEIPAssociate_basic
=== PAUSE TestAccEIPAssociate_basic
=== CONT  TestAccEIPAssociate_basic
--- PASS: TestAccEIPAssociate_basic (234.05s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/eip       234.188s
```

**Pull request merged**